Tests/Integration/Components/CategoryResolverTest.php

```diff
@@ -0,0 +1,101 @@
+        $test1ID = $this->manager->getConnection()->executeQuery('SELECT id FROM s_plugin_connect_categories WHERE category_key = "/deutsch/test1"')->fetchColumn();
+        $test2ID = $this->manager->getConnection()->executeQuery('SELECT id FROM s_plugin_connect_categories WHERE category_key = "/deutsch/test2"')->fetchColumn();
+
+
+        $categories = [
+            '/deutsch' => 'Deutsch',
+            '/deutsch/test1' => 'Test 1',
+            '/deutsch/test3' => 'Test 3',
+            '/deutsch/test3/test31' => 'Test 3.1'
+        ];
+
+        $this->categoryResolver->storeRemoteCategories($categories, 3);
+
+        $actualGermanID = $this->manager->getConnection()->executeQuery('SELECT id FROM s_plugin_connect_categories WHERE category_key = "/deutsch"')->fetchColumn();
+        $actualTest1ID = $this->manager->getConnection()->executeQuery('SELECT id FROM s_plugin_connect_categories WHERE category_key = "/deutsch/test1"')->fetchColumn();
+        $actualTest2ID = $this->manager->getConnection()->executeQuery('SELECT id FROM s_plugin_connect_categories WHERE category_key = "/deutsch/test2"')->fetchColumn();
+
+        //Assert that old remote categories aren't changed
+        $this->assertEquals($germanID, $actualGermanID);
+        $this->assertEquals($test1ID, $actualTest1ID);
+        $this->assertEquals($test2ID, $actualTest2ID);
+
+        $test3ID = $this->manager->getConnection()->executeQuery('SELECT id FROM s_plugin_connect_categories WHERE category_key = "/deutsch/test3"')->fetchColumn();
+        $test31ID = $this->manager->getConnection()->executeQuery('SELECT id FROM s_plugin_connect_categories WHERE category_key = "/deutsch/test3/test31"')->fetchColumn();
+
+        //Assert that new Categories are created
+        $this->assertInternalType('string', $test3ID);
+        $this->assertInternalType('string', $test31ID);
+
+        $productToCategoryID = $this->manager->getConnection()->executeQuery("SELECT id FROM s_plugin_connect_product_to_categories WHERE articleID = 3 AND connect_category_id = $test1ID")->fetchColumn();
+
+        //Assert that old category is still assigned
+        $this->assertInternalType('string', $productToCategoryID);
+
+        $productToCategoryID = $this->manager->getConnection()->executeQuery("SELECT id FROM s_plugin_connect_product_to_categories WHERE articleID = 3 AND connect_category_id = $test2ID")->fetchColumn();
+
+        //Assert that removed category is not assigned
+        $this->assertEquals(false, $productToCategoryID);
+
+        $productToCategoryID = $this->manager->getConnection()->executeQuery("SELECT id FROM s_plugin_connect_product_to_categories WHERE articleID = 3 AND connect_category_id = $test3ID")->fetchColumn();
+
+        //Assert that new, not-leaf category is assigned
+        //This is necessary that ext.js find all products in not-leaf categories in import window
+        $this->assertInternalType('string', $productToCategoryID);
```

Comment:
I think it looks better to check that ```$productToCategoryID``` is greater than zero.

Reply:
but zero is a valid ID or not?
then you have to cast all IDs returned from executeQuery() to int and that would crash if no result is found becuase then executeQuery returns false